### PR TITLE
feat: add support to preferAuthorize intent

### PR DESCRIPTION
**Pull request by brunodesde1987**
# Description

Adds support to the new `preferAuthorize` intent.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own changes
- [x] I have run `yarn lint` to make sure my changes pass all tests
- [x] I have run `yarn test` to make sure my changes pass all linters
- [x] I have pulled the latest changes from the upstream `main` branch
- [ ] I have tested both the react and the CDN versions on local and integration environments
- [ ] I have added the necessary labels to this PR in case a new release needs to be published after merging into `main` (e.g. `release` and `patch`)

## Contribution guidelines

For contribution guidelines, styleguide, and other helpful information please
see the `CONTRIBUTING.md` file in the root of this project.

### CODE REVIEW

File: packages/embed/src/validation.test.ts [hunk 1]
```diff
@@ -609,15 +609,15 @@ describe('validateIntent()', () => {
...
+    ;['authorize', 'preferAuthorize', 'capture'].forEach((value) => {
       const options = {
         ...defaultOptions,
         value,
         callback: jest.fn(),
       }
       const valid = validateIntent({
         ...defaultOptions,
-        value: 'authorize',
+        value,
```
Comment (brunodesde1987) [inline packages/embed/src/validation.test.ts:620]: This hardcoded `'authorize'` seems to be a mistake. Is that right? @luca-gr4vy

Comment (luca-gr4vy) [reply packages/embed/src/validation.test.ts:620]: yeah, probably. I see it should use values from the forEach array

**Review by luca-gr4vy — APPROVED** [review]
Looks good 👍 don't forget to add the `release` and `minor` (probably good in this case instead of `patch`) labels

**Comment by gr4vy-code** [comment]
:rocket: PR was released in [`v2.32.0`](https://github.com/gr4vy/gr4vy-embed/releases/tag/v2.32.0) :rocket:

**Pull request merged**